### ISSUE TITLE
iframe HTML content cannot scroll on iOS browsers

### ISSUE THREAD

**Issue by Hodgekins** [Scirra/Construct-3-bugs#2435]
## Problem description

I'm using the iframe object to display html content.

Users are unable to scroll though content displayed in an iframe on iPad and iPhone browsers (also affects Xcode exports)

Users are able to scroll on Chrome and Safari desktop version.
Users are able to scroll in Android apps.

## Attach a .c3p

[iframe scrolling example.zip](https://github.com/Scirra/Construct-3-bugs/files/2667555/iframe.scrolling.example.zip)

## Steps to reproduce

1. Select remote preview.
2. Load on iOS device.

or 

1. Export as Xcode project.
2. Test on iOS device.

I don't have an iPhone so I've only been able to test on an iPad. My friend remotely tested on her iPhone and said she observed the same result (she couldn't scroll).

## Observed result

User is unable to scroll iframe content.

## Expected result

User to be able to scroll vertically to see all content in iframe.

## First affected release

r129

## Affected browsers/platforms

Only observed issue on iOS devices.

Let me know if you'd like me to provide any more info. I'm really sorry if I've missed something obvious!

**Comment by shortercode** [comment]
iOS doesn't conform to standard behaviour for iframe sizing, so I expect this is a side effect of that or how we have worked around it. Assigning to @AshleyScirra to review.

**Comment by AshleyScirra** [comment]
Yeah, iOS iframes have always been buggy, no idea why they've never fixed it... IIRC you can have either:

- a fixed size, and no scrolling, or
- a variable size, extending to the size of the content, with scrolling

Neither is a particularly good option for the iframe plugin...

**Comment by Hodgekins** [comment]
I've had a limited amount of success making a page on my website with this html:

```
<div style="height:500px; -webkit-overflow-scrolling: touch; overflow-y: scroll;">
  <iframe src="https://construct.net" style="width:100%;height:100%">Construct 3</iframe>
</div>
```
This displays on an iPad and you can scroll, although it seems to resize in weird ways.

I followed advice here: https://davidwalsh.name/scroll-iframes-ios

I've been less successful making this work in Construct, but this is how far I've got:

[iframe test 2.zip](https://github.com/Scirra/Construct-3-bugs/files/2668682/iframe.test.2.zip)

(This uses Valery Popov's Javascript plugin)

**Comment by AshleyScirra** [comment]
Please note our bug report guidelines require a Construct 3 project not using any third-party addons. We can't do anything with exported projects. Since no project is provided I will be testing with this one: [iframe-test.zip](https://github.com/Scirra/Construct-3-bugs/files/2773040/iframe-test.zip)

**Comment by AshleyScirra** [comment]
I can't figure out a way to get this to work reasonably, even with the provided hacks. I tried the default construct.net page with the wrapper element with those styles. It has two problems:

1. The page adopts a full-size viewport, ignoring any responsive design - so it looks different to the current approach (straight away creating a compatibility problem), and the page goes so huge it's not really useful to use for reading anything
2. The page can be scrolled, but it flickers horribly while scrolling and keeps jumping back up to the top-left corner

Basically I think iframes on iOS are just too broken to be able to support! So I think we have to keep our workaround, and you'll either just have to wait for Apple to fix their iframe support, or implement your own hacks in JS. At least non-scrollable iframes still have uses for things like video embeds.
